### PR TITLE
Expose args for blending, visible, opacity

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR performs a little refactoring around the visual wrapper for the layers, including exposing `opacity`, `blending`, and `visible` as three new keyword arguments for every layer. I found myself often wanting to pass these directly to the `add_image`, `add_points` etc. functions and now you can. I've also added doc strings and tests for all these properties. No functionality should change.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] add new tests to each layer model, and add `examples/pass_colormaps.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/layers/base/base.py [hunk 1]
```diff
@@ -20,6 +20,85 @@ class Layer(VisualWrapper, KeymapMixin, ABC):
...
         from `cls._basename()`
+    opacity : float
+        Opacity of the layer visual, between 0.0 and 1.0.
+    blending : str
+        One of a list of preset blending modes that determines how RGB and
+        alpha values of the layer visual get mixed. Allowed values are
+        {'opaque', 'translucent', and 'additive'}.
+    visible : bool
+        Whether the layer visual is currently being displayed.
+
```
Comment (kne42) [inline napari/layers/base/base.py:31]: you should only have 1 newline here

File: napari/layers/base/base.py [hunk 2]
```diff
@@ -20,6 +20,85 @@ class Layer(VisualWrapper, KeymapMixin, ABC):
...
+        Determine if canvas pan/zoom interactivity is enabled.
+    cursor : str
+        String identifying which cursor displayed over canvas.
+    cursor_size : int | None
+        Size of cursor if custom. None yields default size
+    scale_factor : float
+        Conversion factor from canvas coordinates to image coordinates, which
+        depends on the current zoom level.
+
+    Extended Summary
```
Comment (kne42) [inline napari/layers/base/base.py:93]: there is no section in the numpy doc format for an "extended summary." see "notes" instead

File: napari/layers/base/base.py [hunk 3]
```diff
@@ -20,6 +20,85 @@ class Layer(VisualWrapper, KeymapMixin, ABC):
...
+    ----------
```
Comment (kne42) [inline napari/layers/base/base.py]: "underline" should match length of corresponding header

File: napari/layers/base/base.py [hunk 4]
```diff
@@ -20,6 +20,85 @@ class Layer(VisualWrapper, KeymapMixin, ABC):
...
+    _master_transform : vispy.visuals.transforms.STTransform
```
Comment (kne42) [inline napari/layers/base/base.py:95]: private attributes/methods should not be included in the docstring

File: napari/layers/image/image.py [hunk 5]
```diff
@@ -105,12 +113,21 @@ def __init__(
         clim=None,
         clim_range=None,
         interpolation='nearest',
+        opacity=1,
```
Comment (kne42) [inline napari/layers/image/image.py:116]: could we make these keyword-only arguments by adding the `*` as a separate argument?

File: napari/layers/labels/labels.py [hunk 6]
```diff
@@ -100,12 +106,20 @@ def __init__(
         num_colors=50,
         seed=0.5,
         opacity=0.7,
```
Comment (kne42) [inline napari/layers/labels/labels.py:108]: see above

File: napari/layers/points/points.py [hunk 7]
```diff
@@ -113,6 +121,9 @@ def __init__(
         edge_color='black',
         face_color='white',
         n_dimensional=False,
+        opacity=1,
```
Comment (kne42) [inline napari/layers/points/points.py:124]: see above

File: napari/layers/shapes/shapes.py [hunk 8]
```diff
@@ -213,8 +219,10 @@ def __init__(
         edge_width=1,
         edge_color='black',
         face_color='white',
-        opacity=0.7,
         z_index=0,
+        opacity=0.7,
```
Comment (kne42) [inline napari/layers/shapes/shapes.py:223]: see above

File: napari/layers/vectors/vectors.py [hunk 9]
```diff
@@ -65,10 +73,25 @@ class Vectors(Layer):
...
 
     def __init__(
-        self, vectors, *, edge_width=1, edge_color='red', length=1, name=None
+        self,
+        vectors,
+        *,
+        edge_width=1,
+        edge_color='red',
+        length=1,
+        opacity=1,
```
Comment (kne42) [inline napari/layers/vectors/vectors.py:82]: see above

File: napari/layers/volume/volume.py [hunk 10]
```diff
@@ -74,12 +82,21 @@ def __init__(
         colormap='gray',
         clim=None,
         clim_range=None,
+        opacity=1,
```
Comment (kne42) [inline napari/layers/volume/volume.py:85]: see above

**Review by kne42 — APPROVED** [review]
would like to see all arguments shared between all layer types as keyword-only

File: napari/layers/base/base.py (hunk 1 above)
Comment (sofroniewn) [reply napari/layers/base/base.py:31]: fixed

File: napari/layers/base/base.py (hunk 2 above)
Comment (sofroniewn) [reply napari/layers/base/base.py:93]: hmm, we've been using "extended summary" in other places, i think at the suggestion of @jni - happy to refactor, but would rather have a separate PR that fixes everywhere

File: napari/layers/base/base.py (hunk 4 above)
Comment (sofroniewn) [reply napari/layers/base/base.py:95]: we'll need to fix this on our other layers too then - i'd rather wait for now and have a separate PR do all the doc strings for all the layers

File: napari/layers/base/base.py (hunk 3 above)
Comment (sofroniewn) [reply napari/layers/base/base.py]: fixed, but we'll need to fix elsewhere too

File: napari/layers/image/image.py (hunk 5 above)
Comment (sofroniewn) [reply napari/layers/image/image.py:116]: we already have a `*` at line 109. why do we need another?

```python
    def __init__(
        self,
        image,
        *,
        metadata=None,
        multichannel=None,
        colormap='gray',
        clim=None,
        clim_range=None,
        interpolation='nearest',
        opacity=1,
        blending='translucent',
        visible=True,
        name=None,
        **kwargs,
    ):
```

Comment (kne42) [reply napari/layers/image/image.py:116]: didn't see that, mb

**Pull request merged**